pkg/solana/logpoller/log_poller.go

```diff
@@ -0,0 +1,316 @@
+
+		filtersByName:    make(map[string]Filter),
+		filtersByAddress: map[PublicKey]map[EventSignature][]Filter{},
+	}
+}
+
+func (lp *LogPoller) Start(context.Context) error {
+	return lp.StartOnce("LogPoller", func() error {
+		lp.wg.Add(2)
+		go lp.run()
+		go lp.backgroundWorkerRun()
+		return nil
+	})
+}
+
+func (lp *LogPoller) Close() error {
+	return lp.StopOnce("LogPoller", func() error {
+		close(lp.chStop)
+		lp.wg.Wait()
+		return nil
+	})
+}
+
+func (lp *LogPoller) run() {
+	defer lp.wg.Done()
+	ctx, cancel := lp.chStop.NewCtx()
+	defer cancel()
+
+	var blocks chan struct {
+		BlockNumber int64
+		Logs        any // to be defined
+	}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case block := <-blocks:
+			lp.filtersMutex.Lock()
+			filtersToBackfill := lp.filtersToBackfill
+			lp.filtersToBackfill = nil
+			lp.filtersMutex.Unlock()
+			// TODO: NONEVM-916 parse, filters and persist logs
+			// NOTE: removal of filters occurs in the separate goroutine, so there is a chance that upon insert
+			// of log corresponding filter won't be present in the db. Ensure to refilter and retry on insert error
+			for _, filter := range filtersToBackfill {
+				go lp.startFilterBackfill(ctx, filter, block.BlockNumber)
+			}
+		}
+	}
+}
+
+func (lp *LogPoller) backgroundWorkerRun() {
+	defer lp.wg.Done()
+	ctx, cancel := lp.chStop.NewCtx()
+	defer cancel()
+
+	pruneFilters := services.NewTicker(time.Minute + 618*time.Millisecond) // try to minimize collisions with one-second period
+	defer pruneFilters.Stop()
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-pruneFilters.C:
+			err := lp.pruneFilters(ctx)
+			if err != nil {
+				lp.lggr.Errorw("Failed to prune filters", "err", err)
+			}
+		}
+	}
+}
+
+func (lp *LogPoller) pruneFilters(ctx context.Context) error {
+	err := lp.loadFilters(ctx)
+	if err != nil {
+		return fmt.Errorf("failed to load filters: %w", err)
+	}
+
+	lp.filtersMutex.Lock()
+	filtersToDelete := lp.filtersToDelete
+	lp.filtersToDelete = nil
+	lp.filtersMutex.Unlock()
+
+	if len(filtersToDelete) == 0 {
+		return nil
+	}
+
+	err = lp.orm.DeleteFilters(ctx, filtersToDelete)
+	if err != nil {
+		lp.filtersMutex.Lock()
+		defer lp.filtersMutex.Unlock()
+		lp.filtersToDelete = append(lp.filtersToDelete, filtersToDelete...)
+		return fmt.Errorf("failed to delete filters: %w", err)
+	}
+
+	return nil
+}
+
+func (lp *LogPoller) startFilterBackfill(ctx context.Context, filter Filter, toBlock int64) {
+	// TODO: NONEVM-916 start backfill
+	lp.lggr.Debugw("Starting filter backfill", "filter", filter)
+}
+
+// RegisterFilter persists provided filter and ensures that any log emitted by a contract with filter.Address
+// that matches filter.EventSig signature will be captured starting from filter.StartingBlock.
+// filter.Name must be unique otherwise ErrFilterNameConflict is returned. // TODO: not sure this is a good idea. Callers are most likely going to ignore this error
```

Comment:
I think what we decided in the most recent standup is that it should behave like evm does after all, right? (Basically, update it instead of inserting a new one... so that if a new version of chainlink is released that has a different retention time on the filter or other properties, those get updated. Also possible that a product might have that determined by toml config, to allow node ops to update it... but AFAIK nobody is doing that yet.)